03 - Git/README.md

```diff
@@ -1 +1,347 @@
+
+```
+git pull
+```
+
+This will check for new changes on your repository and incorporate them to your current branch.
+
+If you want a list of the commits present on branch master you can use:
+
+```
+git log
+```
+
+Then, using the commit hash, you can see more information using
+
+```
+git show <hash>
+```
+
+This will show information like new files, deleted files, new added code, removed code that were modified with that commit.
+
+## Branches
+
+When working on a project, there are going to be many different features to add, bugs to fix, or even new ideas to try. Changes that could potentially break other functionality. We use branching to help us with this issues.
+When creating a branch, we’re basically copying our own version of the project, creating an environment to test out new ideas and make changes that don’t affect the master branch.
+
+To create a branch, simply run this command on your terminal
+
+```
+git branch <branch_name>
+```
+_***<branch_name>: A significant name about what you’re working on. Example: story-123-fixing-payment-issues***_
+
+After you create it, you must move to the that branch before starting to work on it. You do so by using command
+
+```
+git checkout <branch_name>
+```
+
+_***If you want to create a branch and check out at the same time you can use `git checkout -b <branch_name>`***_
+
+Now we're ready to start working on our new feature
+
+```
+Exercise:
+Change your teammate "Hello, World" implementation to any other phrase you want and then commit and push your changes.
+```
+
+If you checkout to branch `master` and then `git log`, You'll notice that your commit is not present there, that's because you committed to another branch.
+
+After you finish working on your feature, it's now ready to be added to the master branch. In order to do so, you need to `merge` your branch to master.
+
+Make sure both of your branches are up to date using `git pull` and then run
+
+```
+git checkout <branch_name>
+git pull
+git checkout master
+git merge <branch_name>
+git push -u origin master
+```
+Now, if you run `git log`, You'll notice that your new commit is there on the `master` branch.
+
+<p align="center">
+<img align="center" height="500" src="https://user-images.githubusercontent.com/7427385/37220229-32b2ca1a-2383-11e8-8519-e481dce6de06.png">
+</p>
+
+## Pull requests
+Pull requests are a way to propose new changes on your project.
+
+Instead of merging directly to your master/development branch. You open a pull request to let other review and approve your changes before merging.
+
+This way, branches become cleaner and easily maintainable.
+
+To perform a pull request (PR for short). Follow the branches tutorial but skip the merging phase. Instead, go to your repository on GitHub and click on _Compare and pull request_.
+
+First of all, you need to select which branch are you going to merge to, in this case select `master`.
```

Comment:
First of all, you need to select which branch ~are you~ you are going to merge to, in this case select `master`.